From 04b1c4f3ce4894bcfc128af2881701a641f7f629 Mon Sep 17 00:00:00 2001
From: "hong.wang" <hong.wang@amlogic.com>
Date: Tue, 26 Apr 2022 10:05:11 +0800
Subject: [PATCH] BT: disable wbs and remove BTA_AG_FEAT_CODEC for hfp.[1/1]

PD#SWPL-74037

Problem:
    The voice call is silent after the Bluetooth headset is connected.

Solution:
    disable wbs and remove BTA_AG_FEAT_CODEC for hfp,audio change something.

Verify: AN400
Change-Id: I68af6672c36918771d095538ae14c17301a13eeb
Signed-off-by: hong.wang <hong.wang@amlogic.com>
---
 bta/ag/bta_ag_act.cc             | 2 ++
 bta/ag/bta_ag_sdp.cc             | 2 ++
 btif/src/btif_hf.cc              | 2 +-
 device/include/esco_parameters.h | 2 +-
 internal_include/bt_target.h     | 2 +-
 5 files changed, 7 insertions(+), 3 deletions(-)
 mode change 100644 => 100755 bta/ag/bta_ag_act.cc
 mode change 100644 => 100755 bta/ag/bta_ag_sdp.cc
 mode change 100644 => 100755 btif/src/btif_hf.cc
 mode change 100644 => 100755 device/include/esco_parameters.h
 mode change 100644 => 100755 internal_include/bt_target.h

diff --git a/bta/ag/bta_ag_act.cc b/bta/ag/bta_ag_act.cc
old mode 100644
new mode 100755
index 5a3697aa2..6e5cd1faf
--- a/bta/ag/bta_ag_act.cc
+++ b/bta/ag/bta_ag_act.cc
@@ -469,12 +469,14 @@ void bta_ag_rfc_open(tBTA_AG_SCB* p_scb, const tBTA_AG_DATA& data) {
     if (btif_config_get_bin(
             p_scb->peer_addr.ToString(), HFP_SDP_FEATURES_CONFIG_KEY,
             (uint8_t*)&p_scb->peer_sdp_features, &sdp_features_size)) {
+      #if (DISABLE_WBS == FALSE)
       bool sdp_wbs_support = p_scb->peer_sdp_features & BTA_AG_FEAT_WBS_SUPPORT;
       if (!p_scb->received_at_bac && sdp_wbs_support) {
         p_scb->codec_updated = true;
         p_scb->peer_codecs = BTA_AG_CODEC_CVSD & BTA_AG_CODEC_MSBC;
         p_scb->sco_codec = UUID_CODEC_MSBC;
       }
+      #endif
     } else {
       APPL_TRACE_WARNING("%s: Failed read cached peer HFP SDP features for %s",
                          __func__, p_scb->peer_addr.ToString().c_str());
diff --git a/bta/ag/bta_ag_sdp.cc b/bta/ag/bta_ag_sdp.cc
old mode 100644
new mode 100755
index dd9ec3c8c..28305746a
--- a/bta/ag/bta_ag_sdp.cc
+++ b/bta/ag/bta_ag_sdp.cc
@@ -365,6 +365,7 @@ bool bta_ag_sdp_find_attr(tBTA_AG_SCB* p_scb, tBTA_SERVICE_MASK service) {
         /* There might be race condition between SDP and BRSF.  */
         /* Do not update if we already received BRSF.           */
         uint16_t sdp_features = p_attr->attr_value.v.u16;
+        #if (DISABLE_WBS == FALSE)
         bool sdp_wbs_support = sdp_features & BTA_AG_FEAT_WBS_SUPPORT;
         if (!p_scb->received_at_bac && sdp_wbs_support) {
           // Workaround for misbehaving HFs (e.g. some Hyundai car kit) that:
@@ -375,6 +376,7 @@ bool bta_ag_sdp_find_attr(tBTA_AG_SCB* p_scb, tBTA_SERVICE_MASK service) {
           p_scb->peer_codecs = BTA_AG_CODEC_CVSD & BTA_AG_CODEC_MSBC;
           p_scb->sco_codec = UUID_CODEC_MSBC;
         }
+        #endif
         if (sdp_features != p_scb->peer_sdp_features) {
           p_scb->peer_sdp_features = sdp_features;
           if (btif_config_set_bin(
diff --git a/btif/src/btif_hf.cc b/btif/src/btif_hf.cc
old mode 100644
new mode 100755
index ec4c13f47..fdc1aa6fc
--- a/btif/src/btif_hf.cc
+++ b/btif/src/btif_hf.cc
@@ -77,7 +77,7 @@ namespace headset {
 #define BTIF_HF_FEATURES                                       \
   (BTA_AG_FEAT_3WAY | BTA_AG_FEAT_ECNR | BTA_AG_FEAT_REJECT |  \
    BTA_AG_FEAT_ECS | BTA_AG_FEAT_EXTERR | BTA_AG_FEAT_VREC |   \
-   BTA_AG_FEAT_CODEC | BTA_AG_FEAT_HF_IND | BTA_AG_FEAT_ESCO | \
+   BTA_AG_FEAT_HF_IND | BTA_AG_FEAT_ESCO | \
    BTA_AG_FEAT_UNAT)
 #endif
 
diff --git a/device/include/esco_parameters.h b/device/include/esco_parameters.h
old mode 100644
new mode 100755
index b79948d00..6154ffc28
--- a/device/include/esco_parameters.h
+++ b/device/include/esco_parameters.h
@@ -57,7 +57,7 @@ typedef uint8_t esco_coding_format_t;
 typedef uint8_t esco_pcm_data_format_t;
 
 // SCO Data Path
-#define ESCO_DATA_PATH_PCM 1                /* 0x01-0xFE (PCM Chan) */
+#define ESCO_DATA_PATH_PCM 6                /* 0x01-0xFE (PCM Chan) */
 #define ESCO_DATA_PATH_HCI ((uint8_t)0x00)  /* HCI-0, 0x01-0xFE (PCM Chan) */
 #define ESCO_DATA_PATH_TEST ((uint8_t)0xFF) /* 0xFF-Audio Test */
 typedef uint8_t esco_data_path_t;
diff --git a/internal_include/bt_target.h b/internal_include/bt_target.h
old mode 100644
new mode 100755
index 4aca86822..c4f29252a
--- a/internal_include/bt_target.h
+++ b/internal_include/bt_target.h
@@ -276,7 +276,7 @@
 #endif
 
 #ifndef DISABLE_WBS
-#define DISABLE_WBS FALSE
+#define DISABLE_WBS TRUE
 #endif
 
 /*  This is used to work around a controller bug that doesn't like Disconnect
-- 
2.35.1

